### ISSUE TITLE
Parse server error code 202 throws ParseException with message 'Empty reply from server'

### ISSUE THREAD

**Issue by nargesaj** [ParsePlatform/parse-server#2754]
Possibly related to issue #1890

I am trying to use the following code from my Parse Server instance on AWS. The success part works fine. It creates new account successfully. But, the error part fails. When I am trying to create a new account using the already exists one, It throws **'Parse\ParseException' with message 'Empty reply from server'** at application side.

Parse.Cloud.define("createAccount", function(request, response) {
//Parse.Cloud.useMasterKey();

var user = new Parse.User();

user.set("Type", request.params.Type);
user.set("name", request.params.name);
user.set("email", request.params.email);
user.set("username", request.params.username);
user.set("password", request.params.password);
user.set("phone", request.params.phone);

user.signUp(null, { useMasterKey: true }).then(
function(user) {
response.success({status: "success", message: "Created Account Successfully", id: user.id});
},
function(user, error) { 
console.log(error);
response.error({status: "Error", message:error.code + " " + error.message});
});
});

verbose: 
error: Error generating response. ParseError {
code: 202,
message: 'Account already exists for this username.' } code=202, message=Account already exists for this username.
undefined
[object Object]

The serverURL variable in index.js is set to:
serverURL: process.env.SERVER_URL || 'http://localhost:1337/parse'

**Comment by flovilmart** [comment]
does the error get log to the console on `console.log(error)`

**Comment by nargesaj** [comment]
It logs 'undefined' to the console.

**Comment by flovilmart** [comment]
I didn't notice: `function(user, error) { ` should be `function(error) {` promises callbacks usually take only one argument :)

**Comment by nargesaj** [comment]
Thank you. But, it then changes the error to 

error: Error generating response. ParseError {
  code: 141,
  message: 
   { status: 'Error',
     message: '202 Account already exists for this username.' } } code=141, status=Error, message=202 Account already exists for this username.

I am running parse-server-example. I was wondering if this error related to my parse server instance?

**Comment by flovilmart** [comment]
Uhm ok...

**Comment by nargesaj** [comment]
Ok. I installed local parse server by using the following steps:
1. Created an empty dir.
2. `$ npm init`
3. `$ npm install express --save`
4. `$ npm install parse-server`
5. Created an `index.js` config file with my options
6. Run `node index.js`

As you mentioned I changed  `function(user, error) {`  to `function(error) {`. 

The success function works fine as before, but the error function still fails. 
```
error: Error generating response. ParseError {
  code: 202,
  message: 'Account already exists for this username.' } code=202, message=Account already exists for this username.
console.log: 202 Account already exists for this username.
error: Failed running cloud function createAccount for user 7jGSuYbQ5k with:
  Input: 
```
It also throws `PHP Fatal error:  Wrong parameters for Exception([string $exception [, long $code [, Exception $previous = NULL]]])` at application side.
